pkg/network/ebpf/c/prebuilt/http.c

```diff
@@ -239,6 +240,65 @@ int uprobe__SSL_shutdown(struct pt_regs* ctx) {
     return 0;
 }
 
+SEC("kprobe/do_sys_open")
+int kprobe__do_sys_open(struct pt_regs* ctx) {
+    char *path_argument = (char *)PT_REGS_PARM2(ctx);
+    lib_path_t path = {0};
+    bpf_probe_read(path.buf, sizeof(path.buf), path_argument);
+
+    // Find the null character and clean up the garbage following it
+#pragma unroll
+    for (int i = 0; i < LIB_PATH_MAX_SIZE; i++) {
+        if (path.len) {
+            path.buf[i] = 0;
+        } else if (path.buf[i] == 0) {
+            path.len = i;
+        }
+    }
+
+    // Bail out if the path size is larger than our buffer
+    if (!path.len) {
+        return 0;
+    }
+
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+    bpf_map_update_elem(&open_at_args, &pid_tgid, &path, BPF_ANY);
+    return 0;
+}
+
+SEC("kretprobe/do_sys_open")
+int kretprobe__do_sys_open(struct pt_regs* ctx) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
```

Comment:
you could check the return value of the open syscall to avoid send event for non existing file

Reply:
Good point 👍